docs/zksync-era-integration.md

```diff
@@ -0,0 +1,105 @@
+## Zksync-era integration
+
+The `era_vm` doesn't do much by itself, it needs someone to tell it what to do and orquestrate it. On ethereum, for example, the execution node would take a block and process all of the transactions within it one by one using the `evm`. In zksync, things are a bit different, here transactions aren't processed one by one but in batch, this is to publish all of the transactions as just a single one, making the l1 post cheaper and distribitued among all the trnansactions withing the block.
+
+This is what the bootloader do. The bootloader is a `.yul` contract that lives on l1 and takes an array of transactions and executes them in the `era_vm`.
+
+Part of integrating our vm with the zk stack comes down to integrate the bootloader with our vm. That is, we need to make our vm "talk" with the bootloader. Before diving into the integration, we'll first explore what the bootloader does more in detail, which will explain what parts we need to keep track on our vm.
```

Comment:
> Part of integrating our vm with the zk stack comes down to integrate the bootloader with our vm. That is, we need to make our vm "talk" with the bootloader.

Couldn't the bootloader just run on top of our VM? I believe the work on the bootloader is more about giving us some flexibility and removing overhead for a "fast run" VM for which a proof is not expected.

Reply:
Btw, there a lot of typos there I am aware I need to fix.